### DISCUSSION TITLE
New LDAP users don't show up

### DISCUSSION THREAD

**Discussion by Saphar** [processone/ejabberd#4319] [Questions]
Hello, I have configured ejabberd with ldap. My Ldap is a local replication of another ldap installation. Whenever I create a new user in LDAP it is not being loaded in Ejabberd, so the user cannot log in.
I can confirm that the ldap replication has the new user.
I tried restarting ejabberd but it doesn't reload the users. I don't know what triggers the user reload, but at some point works after few days.

**Comment by badlop** [comment]
> Hello, I have configured ejabberd with ldap. My Ldap is a local replication of another ldap installation.

I guess you are using the latest stable ejabberd release, installed using an official method (source code, binary installer or container image), have not setup any additional module, and no substantial configuration changes. And you have setup something like this in ejabberd configuration file:

```yaml
auth_method: [ldap]
```

> Whenever I create a new user in LDAP it is not being loaded in Ejabberd, so the user cannot log in.

What lines appear in the ejabberd log file when the login is attempted?

I added a user to my LDAP test server, and the account was immediately listed by ejabberd, and it can login.

> I can confirm that the ldap replication has the new user.

Are the new accounts shown by the `registered_users` command? You can use the command line or WebAdmin, for example:

```bash
$ ejabberdctl registered_users localhost
admin
test6000
test6001
test6002
```

> I tried restarting ejabberd but it doesn't reload the users. I don't know what triggers the user reload, but at some point works after few days.

Try disabling [the option `auth_use_cache`](https://docs.ejabberd.im/admin/configuration/authentication/#general-options), it stores in ejabberd's cache the authentication success and failures for 1 hour.

**Comment by Saphar** [comment]
@badlop Thank you so much for responding and helping us.

> I guess you are using the latest stable ejabberd release, installed using an official method (source code, binary installer or container image), have not setup any additional module, and no substantial configuration changes. And you have setup something like this in ejabberd configuration file: `auth_method: [ldap]`
Sorry I forgot to describe the setup. Yes, that is my auth method.

`ejabberd 24.10.0 is running in that node`

> What lines appear in the ejabberd log file when the login is attempted?
>I added a user to my LDAP test server, and the account was immediately listed by ejabberd, and it can login.

Failed c2s PLAIN authentication for user@domain from IP: Invalid username or password

> Are the new accounts shown by the registered_users command? You can use the command line or WebAdmin, for example:

No, I don't find the respective user.

> Try disabling [the option auth_use_cache](https://docs.ejabberd.im/admin/configuration/authentication/#general-options), it stores in ejabberd's cache the authentication success and failures for 1 hour.

I added: `auth_user_cache: false` and restarted ejabberd, but I still don't find the new user and I am not able to login with it.

`ldapsearch -x -D "cn=admin,dc=domain,dc=ltd" -W -H ldap://127.0.0.1 -b "ou=users,ou=accounts,dc=domain,dc=ltd" -s sub 'uid=myuser'`

It is returning the user from ldap, so I know it's there.
